### PR TITLE
Refactoring and linter warnings fixes

### PR DESCRIPTION
**Pull request by artemgavrilov**
Refactoring, documentation and linter fixes

FB: https://github.com/Percona-Lab/pmm-submodules/pull/913

### CODE REVIEW

File: models/settings_helpers.go [hunk 1]
```diff
@@ -139,10 +139,10 @@ func UpdateSettings(q reform.DBTX, params *ChangeSettingsParams) (*Settings, err
 // ValidateSettings validates settings changes.
 func ValidateSettings(params *ChangeSettingsParams) error {
 	if params.EnableTelemetry && params.DisableTelemetry {
-		return fmt.Errorf("Both enable_telemetry and disable_telemetry are present.")
+		return errors.New("Both enable_telemetry and disable_telemetry are present.")
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply models/settings_helpers.go:142]: The text used in gRPC errors.

File: models/settings_helpers.go [hunk 2]
```diff
@@ -139,10 +139,10 @@ func UpdateSettings(q reform.DBTX, params *ChangeSettingsParams) (*Settings, err
...
 	}
 	if params.EnableSTT && params.DisableSTT {
-		return fmt.Errorf("Both enable_stt and disable_stt are present.")
+		return errors.New("Both enable_stt and disable_stt are present.")
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply models/settings_helpers.go:145]: The text used in gRPC errors.

File: models/settings_helpers.go [hunk 3]
```diff
@@ -190,7 +190,7 @@ func ValidateSettings(params *ChangeSettingsParams) error {
 
 	if params.AlertManagerURL != "" {
 		if params.RemoveAlertManagerURL {
-			return fmt.Errorf("Both alert_manager_url and remove_alert_manager_url are present.")
+			return errors.New("Both alert_manager_url and remove_alert_manager_url are present.")
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply models/settings_helpers.go:193]: The text used in gRPC errors.

File: models/settings_helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlekSi) [reply models/settings_helpers.go:142]: `errors.New` creates an error with stack trace. We don't want it there.

File: services/prometheus/prometheus.go [hunk 4]
```diff
@@ -129,7 +129,7 @@ func (svc *Service) reload() error {
 	}
 	defer resp.Body.Close() //nolint:errcheck
 
-	if resp.StatusCode == 200 {
+	if resp.StatusCode == 200 { //nolint:gomnd
```
Comment (percona-csalguero) [inline services/prometheus/prometheus.go:132]: Instead of this we could use http.StatusOK

File: services/prometheus/prometheus.go [hunk 5]
```diff
@@ -527,7 +534,7 @@ func (svc *Service) IsReady(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if resp.StatusCode != 200 {
+	if resp.StatusCode != 200 { //nolint:gomnd
```
Comment (percona-csalguero) [inline services/prometheus/prometheus.go:537]: We could change the 200 by http.StatusOK

File: services/prometheus/prometheus.go (hunk 4 above)
Comment (artemgavrilov) [reply services/prometheus/prometheus.go:132]: I had a discussion about that with AlekSi a few times. His point that `200` easier and more native (or something like that) and `http.StatusOK` is redundant. I do not fully agree, but follow common approaches in our codebase.

File: models/settings_helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply models/settings_helpers.go]: Reverted

File: .golangci.yml [hunk 6]
```diff
@@ -28,6 +28,7 @@ linters:
   enable-all: true
   disable:
     - scopelint         # too many false positives
+    - gomnd             # we are using numbers in many cases
```
Comment (percona-csalguero) [inline .golangci.yml:31]: Thanks for this

Comment (artemgavrilov) [reply .golangci.yml:31]: I'm glad you like it 😊

File: utils/clean/clean.go [hunk 7]
```diff
@@ -27,25 +27,25 @@ import (
 	"github.com/percona/pmm-managed/models"
 )
 
-// CleanResults has unexported fields for the results cleanup function.
-type CleanResults struct {
+// Service has unexported fields for the results cleanup function.
+type Service struct {
```
Comment (AlekSi) [inline utils/clean/clean.go:31]: But it is not what we call Service in pmm-managed, that's why it is in utils/clean package.
Let's revert this change to make PR smaller

File: services/prometheus/prometheus.go [hunk 8]
```diff
@@ -346,46 +346,17 @@ func (svc *Service) marshalConfig() ([]byte, error) {
...
-			ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
-				StaticConfigs: []*config.Group{{
-					Targets: []string{"127.0.0.1:9093"},
-				}},
-			},
-			Scheme:     "http",
-			PathPrefix: "/alertmanager/",
-			APIVersion: config.AlertmanagerAPIVersionV2,
-		})
+		conf, err := svc.parseAlertmanagerConfigFromURL("http://127.0.0.1:9093/alertmanager/")
```
Comment (AlekSi) [inline services/prometheus/prometheus.go:349]: Let's also revert this change as it seems to be changing logic, and generally bigger that other changes

File: utils/clean/clean.go (hunk 7 above)
Comment (artemgavrilov) [reply utils/clean/clean.go]: Reverted.

File: services/prometheus/prometheus.go (hunk 8 above)
Comment (artemgavrilov) [reply services/prometheus/prometheus.go]: Reverted.
